src/resmom/linux/alps.c

```diff
@@ -6892,7 +6892,7 @@ create_vnodes_KNL(basil_response_query_system_t *sys_knl)
 		 * If nidlist is empty, node_count gets set to 0 and vnode
 		 * creation in the inner for() is bypassed.
 		 */
-		nid_arr = process_nodelist_KNL(node_group->nidlist, &node_count);
+		nid_arr = (long *) process_nodelist_KNL(node_group->nidlist, &node_count);
```

Comment:
There is no harm in having this change but since you have changed the return value of the function, you don't need it anymore.